firedrake/assemble.py

```diff
@@ -513,7 +520,10 @@ def _collect_lgmaps(matrix, all_bcs, Vrow, Vcol, row, col):
     clgmap = Vcol[col].local_to_global_map(bccol, lgmap=clgmap)
```

Comment:
EquationBCs are the only ones that need to be treated specially for `is_adjoint`, where are they handled?